### PR TITLE
Add md5sum check

### PR DESCRIPTION
**Pull request by squiddle**
this is a cleaned up version of https://github.com/maestrodev/puppet-wget/pull/7

### CODE REVIEW

File: manifests/fetch.pp [hunk 1]
```diff
@@ -110,6 +112,16 @@
     default => undef,
   }
 
+  case $sourceHash {
+    '', undef: {
+      $command = "wget ${verbose_option}${nocheckcert_option}${no_cookies_option}${header_option}${user_option}${output_option}${flags_joined} '${source}'"
```
Comment (carlossg) [inline manifests/fetch.pp]: $command is never used

File: manifests/fetchifhashchanged.pp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+################################################################################
+define wget::fetchifhashchanged(
+  $sourceHash,
+  $destination,
+  $source           = $title,
+  $timeout          = '0',
+  $verbose          = false,
+) {
+
+  exec { "rm ${destination} ; echo ignoredresult":
```
Comment (carlossg) [inline manifests/fetchifhashchanged.pp]: should just be `rm -f` ?

**Comment by carlossg** [comment]
I've added some comments. It'd be good to

* add unit and system tests
* squash the commits into 2: one for implementing retries and another for the sourceHash
* add docs

I would implement the hash change into the same definition instead of creating a new one

**Comment by squiddle** [comment]
- i added unit and system tests
- i removed the retries change, as i do not need it, it was just part of the original pull request
- added additional documentation for $source_hash
- the removal of $destination if $source_hash does not match is now always done, i could not think of why puppet should keep a file, where it knows it is not what is inteded.
- first time i used rebase and push --force for already published changes, feels exciting  :) i hope this is correct now

**Comment by squiddle** [comment]
- i also fixed the $command issue you mentioned
- and i changed the removal operation to use `test`; feels better than a forced rm

**Pull request merged**